src/v2/guide/reactivity.md

```diff
@@ -118,3 +118,7 @@ Vue.component('example', {
   }
 })
 ```
+
+<p class="tip">`$nextTick` relies on native Promises, so heavy operations in a `$nextTick` callback can be render blocking. This is because the callback is pushed into the browsers microtask queue rather then the event loops normal queue. For this reason, when you have heavy synchronized tasks it’s wise to avoid using `$nextTick`, or rather, combine it with a `setTimeout()`. A `setTimeout()` will run UI updates before the callback is picked up by the event loop.
```

Comment:
> `$nextTick` relies on native Promises
> the callback is pushed into the browsers microtask queue

This isn't _quite_ accurate. In 2.4+, `$nextTick` now tries to avoid the microtask queue, currently trying to use:

- `setImmediate` if available, then
- `MessageChannel` if available, then
- `Promise.resolve` (which uses microtasks) if available, then
- `setTimeout`

From the source:

``` js
// An asynchronous deferring mechanism.
// In pre 2.4, we used to use microtasks (Promise/MutationObserver)
// but microtasks actually has too high a priority and fires in between
// supposedly sequential events (e.g. #4521, #6690) or even between
// bubbling of the same event (#6566). Technically setImmediate should be
// the ideal choice, but it's not available everywhere; and the only polyfill
// that consistently queues the callback after all DOM events triggered in the
// same loop is by using MessageChannel.
/* istanbul ignore if */
if (typeof setImmediate !== 'undefined' && isNative(setImmediate)) {
  timerFunc = function () {
    setImmediate(nextTickHandler);
  };
} else if (typeof MessageChannel !== 'undefined' && (
  isNative(MessageChannel) ||
  // PhantomJS
  MessageChannel.toString() === '[object MessageChannelConstructor]'
)) {
  var channel = new MessageChannel();
  var port = channel.port2;
  channel.port1.onmessage = nextTickHandler;
  timerFunc = function () {
    port.postMessage(1);
  };
} else
/* istanbul ignore next */
if (typeof Promise !== 'undefined' && isNative(Promise)) {
  // use microtask in non-DOM environments, e.g. Weex
  var p = Promise.resolve();
  timerFunc = function () {
    p.then(nextTickHandler);
  };
} else {
  // fallback to setTimeout
  timerFunc = function () {
    setTimeout(nextTickHandler, 0);
  };
}
```